src/docfx/build/legacy/LegacyExtensions.cs

```diff
@@ -27,12 +27,12 @@ public static string ToLegacyOutputPathRelativeToBaseSitePath(this Document doc,
         public static string ToLegacySiteUrlRelativeToBaseSitePath(this Document doc, Docset docset)
         {
             var legacyOutputFilePathRelativeToSiteBasePath = doc.SiteUrl;
-            if (legacyOutputFilePathRelativeToSiteBasePath.StartsWith($"/{docset.Config.SiteBasePath}", StringComparison.Ordinal))
+            if (legacyOutputFilePathRelativeToSiteBasePath.StartsWith($"/{docset.Config.SiteBasePath}", PathUtility.PathComparison))
             {
                 legacyOutputFilePathRelativeToSiteBasePath = Path.GetRelativePath(docset.Config.SiteBasePath, doc.SiteUrl.Substring(1));
             }
 
-            return PathUtility.NormalizeFile(doc.FilePath.EndsWith("index.md", StringComparison.OrdinalIgnoreCase) ? $"{legacyOutputFilePathRelativeToSiteBasePath}index" : legacyOutputFilePathRelativeToSiteBasePath);
```

Comment:
>OrdinalIgnoreCase [](start = 96, length = 17)

this one should always ignore case?

Reply:
we need a strategy for handling special file names like _TOC_, _index_, _docfx_. This PR always uses case insensitive comparison for these special files, but we could use os specific comparison.

Reply:
I'm leaning towards using os specifc comparison for these special file names. Sometimes we need to probe files like TOC.md, index.md, this is problematic on linux if it is case insensitive.

Reply:
makes sense. let's go with OS specific

---
In reply to: [200537310](https://github.com/dotnet/docfx/pull/3004#discussion_r200537310) [](ancestors = 200537310)